drivers/SmartThings/zigbee-switch/src/bad_on_off_data_type/init.lua

```diff
@@ -0,0 +1,45 @@
+-- Copyright 2023 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+local data_types = require "st.zigbee.data_types"
+local capabilities = require "st.capabilities"
+
+-- There are reports of at least one device (SONOFF 01MINIZB) which occasionally
+-- reports this value as an Int8, rather than a Boolean, as per the spec
+local function incorrect_data_type_detected(opts, driver, device, zb_rx, ...)
+  return opts.dispatcher_class == "ZigbeeMessageDispatcher" and
+    zb_rx.body.zcl_body and
+    zb_rx.body.zcl_body.attr_record and
+    zb_rx.body.zcl_body.attr_record.data_type.value ~= data_types.Boolean.ID
```

Comment:
Should this be further scoped to make sure it's only applying this logic to the OnOff attribute or does that happen automatically because it's only registering the one attribute handler?

Reply:
@varzac can chime in, but my thinking was that, ultimately, it ends up being the same number of comparisons either way?

Reply:
I would lean toward making this more restrictive to just the onoff attribute just because I think the can handle will be a bit clearer in the actual intent.  But otherwise that restriction will come with the handler check.